### PR TITLE
Move StackdriverTaskHandler to the provider package

### PR DESCRIPTION
**Pull request by ephraimbuddy**
---
This PR fixes one of the issues listed in #9386

The `StackdriverTaskHandler` class from `airflow.utils.log.stackdriver_task_handler` was moved to
`airflow.providers.google.cloud.log.stackdriver_task_handler`.

Make sure to mark the boxes below before creating PR: [x]

- [x] Description above provides context of the change
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [x] Target Github ISSUE in description if exists
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

**Comment by turbaszek** [comment]
@ephraimbuddy can you please take a look at the CI errors?

File: airflow/utils/log/stackdriver_task_handler.py [hunk 1]
```diff
@@ -15,281 +15,14 @@
...
-        return {
-            cls.LABEL_TASK_ID: ti.task_id,
-            cls.LABEL_DAG_ID: ti.dag_id,
-            cls.LABEL_EXECUTION_DATE: str(ti.execution_date.isoformat()),
-            cls.LABEL_TRY_NUMBER: str(ti.try_number),
-        }
+warnings.warn(
+    "This module is deprecated. Please use `airflow.providers.google.cloud.log.stackdriver_task_handler`.",
+    DeprecationWarning, stacklevel=2
+)
```
Comment (turbaszek) [inline airflow/utils/log/stackdriver_task_handler.py:28]: Should we add test in `test_core_to_contrib.py`?

Comment (ephraimbuddy) [reply airflow/utils/log/stackdriver_task_handler.py:28]: Wow. Have been thinking of what to test and how this should be tested. Thanks. Will do it now

**Comment by ephraimbuddy** [comment]
These changes have been move to #9604

**Pull request closed without merging**